wiki/development/coding-guidelines.md

```diff
@@ -293,6 +559,76 @@ As you can see above working with hashlists are fairly simple, a more in depth e
 |`HASHLIST_SELECT(hashlist, index)` | returns the hash at that index in the list. |
 |`HASHLIST_SET(hashlist, index, hash)` | sets a specific index to that hash.|
 
-#### 6.1.1 A note on pass by reference and hashes
+##### 7.4.1.1. A note on pass by reference and hashes
 
 Hashes and hashlists are implemented with SQF arrays, and as such they are passed by reference to other functions. Remember to make copies (using the + operator) if you intend for the hash or hashlist to be modified with out the need for changing the original value.
+
+## 8. Performance Considerations
+
+### 8.1. Adding Elements to Arrays
+When adding new elements to an array, pushback shall be used instead of the binary addition or set. When adding multiple elements to an array `append` may be used instead of pushback.
+
+Good:
+```
+_a pushback _value;
+```
+Also good:
+```
+_a append [1,2,3];
+```
+Bad:
+```
+_a set [ count _a, _value];
+_a = a + _[value];
+```
+When adding an new element to a dynamic location in an array or when the index is precaluclated, set may be used.
+
+When adding multiple elements to an array, the binary addition may be used for the entire addition.
+
+### 8.2. CreateVehicle
+CreateVehicle array shall be used instead of createVehicle
+
+### 8.3. Unscheduled vs Scheduled
+All code that has a visable effect for the user or requires time specific guaranteed excution shall be written in unscheduled space.
+
+### 8.4. Avoid Spawn & execVM
+ExecVM and spawn are to be avoided wherever possible.
+
+### 8.5. Empty Arrays
+When checking if an array is empty, you can use both count or isEqualTo.
```

Comment:
Shouldn't we encourage the use of `isEqualTo`? Unless storing the count in a variable for some other purpose